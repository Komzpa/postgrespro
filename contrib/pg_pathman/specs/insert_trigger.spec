setup
{
	--SELECT '>>> SETUP <<<';
	CREATE EXTENSION pg_pathman;
	CREATE TABLE range_rel(id serial primary key);
	SELECT create_range_partitions('range_rel', 'id', 1, 100, 1);
}

teardown
{
	--SELECT '>>> TEARDOWN <<<';
	--SELECT drop_range_partitions('range_rel');
	DROP TABLE range_rel CASCADE;
	DROP EXTENSION pg_pathman;
}

session "s1"
step "s1i" { INSERT INTO range_rel SELECT generate_series(1, 150); }
step "s1d" { SELECT * FROM pg_inherits WHERE inhparent = 'range_rel'::regclass::oid; }

session "s2"
step "s2i" { INSERT INTO range_rel SELECT generate_series(151, 300); }
