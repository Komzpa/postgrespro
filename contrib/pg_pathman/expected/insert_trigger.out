Parsed test spec with 2 sessions

starting permutation: s1b s1_insert_150 s1r s1_show_partitions s2b s2_insert_150 s2c s2_show_partitions
create_range_partitions

1              
step s1b: BEGIN;
step s1_insert_150: INSERT INTO range_rel SELECT generate_series(1, 150);
step s1r: ROLLBACK;
step s1_show_partitions: SELECT * FROM pg_inherits WHERE inhparent = 'range_rel'::regclass::oid;
inhrelid       inhparent      inhseqno       

16453          16445          1              
16462          16445          1              
step s2b: BEGIN;
step s2_insert_150: INSERT INTO range_rel SELECT generate_series(1, 150);
step s2c: COMMIT;
step s2_show_partitions: SELECT * FROM pg_inherits WHERE inhparent = 'range_rel'::regclass::oid;
inhrelid       inhparent      inhseqno       

16453          16445          1              
16462          16445          1              

starting permutation: s1b s1_insert_150 s1r s1_show_partitions s2b s2_insert_300 s2c s2_show_partitions
create_range_partitions

1              
step s1b: BEGIN;
step s1_insert_150: INSERT INTO range_rel SELECT generate_series(1, 150);
step s1r: ROLLBACK;
step s1_show_partitions: SELECT * FROM pg_inherits WHERE inhparent = 'range_rel'::regclass::oid;
inhrelid       inhparent      inhseqno       

16537          16529          1              
16545          16529          1              
step s2b: BEGIN;
step s2_insert_300: INSERT INTO range_rel SELECT generate_series(151, 300);
step s2c: COMMIT;
step s2_show_partitions: SELECT * FROM pg_inherits WHERE inhparent = 'range_rel'::regclass::oid;
inhrelid       inhparent      inhseqno       

16537          16529          1              
16545          16529          1              
16552          16529          1              

starting permutation: s1b s1_insert_300 s1r s1_show_partitions s2b s2_insert_150 s2c s2_show_partitions
create_range_partitions

1              
step s1b: BEGIN;
step s1_insert_300: INSERT INTO range_rel SELECT generate_series(151, 300);
step s1r: ROLLBACK;
step s1_show_partitions: SELECT * FROM pg_inherits WHERE inhparent = 'range_rel'::regclass::oid;
inhrelid       inhparent      inhseqno       

16627          16619          1              
16635          16619          1              
16642          16619          1              
step s2b: BEGIN;
step s2_insert_150: INSERT INTO range_rel SELECT generate_series(1, 150);
step s2c: COMMIT;
step s2_show_partitions: SELECT * FROM pg_inherits WHERE inhparent = 'range_rel'::regclass::oid;
inhrelid       inhparent      inhseqno       

16627          16619          1              
16635          16619          1              
16642          16619          1              

starting permutation: s1b s1_insert_150 s2b s2_insert_300 s1r s2r s2_show_partitions
create_range_partitions

1              
step s1b: BEGIN;
step s1_insert_150: INSERT INTO range_rel SELECT generate_series(1, 150);
step s2b: BEGIN;
step s2_insert_300: INSERT INTO range_rel SELECT generate_series(151, 300);
step s1r: ROLLBACK;
step s2r: ROLLBACK;
step s2_show_partitions: SELECT * FROM pg_inherits WHERE inhparent = 'range_rel'::regclass::oid;
inhrelid       inhparent      inhseqno       

16717          16709          1              
16725          16709          1              
16732          16709          1              
